### PR TITLE
add change detection for skyThemeIf

### PR DESCRIPTION
**Pull request by johnhwhite**
- when initializing the embedded view, trigger change detection

### CODE REVIEW

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/skyux-theme?_sp.spa=pr-skythemeif-change-detection](https://developer.blackbaud.com/skyux-theme?_sp.spa=pr-skythemeif-change-detection)

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/skyux-theme?_sp.spa=pr-skythemeif-change-detection](https://developer.blackbaud.com/skyux-theme?_sp.spa=pr-skythemeif-change-detection)

File: src/app/public/theming/theme-if.directive.ts [hunk 1]
```diff
@@ -72,11 +76,14 @@ export class SkyThemeIfDirective implements OnDestroy {
   private updateView(): void {
     const condition = this.context && this.currentTheme?.theme.name === this.context;
     if (condition && !this.hasView) {
-      this.viewContainer.createEmbeddedView(this.templateRef);
       this.hasView = true;
+      this.embeddedView = this.viewContainer.createEmbeddedView(this.templateRef);
```
Comment (Blackbaud-SteveBrush) [inline src/app/public/theming/theme-if.directive.ts:80]: Do we still need the `EmbeddedViewRef`?

**Review by Blackbaud-SteveBrush — COMMENTED** [review]
Everything looks good; one comment regarding `EmbeddedViewRef`.

File: src/app/public/theming/theme-if.directive.ts (hunk 1 above)
Comment (johnhwhite) [reply src/app/public/theming/theme-if.directive.ts:80]: I'm using the ref to call `this.embeddedView.destroy();` when the condition flips.

**Review by Blackbaud-SteveBrush — APPROVED** [review]

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/skyux-theme?_sp.spa=pr-skythemeif-change-detection](https://developer.blackbaud.com/skyux-theme?_sp.spa=pr-skythemeif-change-detection)

**Pull request merged**